src/contributor-guide/dod/tco-analysis.md

```diff
@@ -0,0 +1,110 @@
+|                                        | Method parameter default value changed                                                                                   | MAJOR |
+|                                        | Method parameter default value added                                                                                     | MINOR |
+|                                        |                                                                                                                          |       |
+| PHP Class (marked with @api)           | Method argument or return value format changed (Example: before - [1, 2, 3], after - ['items' => [1,2,3], 'count' => 3]) | MAJOR |
+|                                        | Trait removed                                                                                                            | MAJOR |
+| Message queue                          | Topic removed                                                                                                            | MAJOR |
+|                                        | Consumer removed                                                                                                         | MINOR |
+|                                        | New topic published                                                                                                      | MINOR |
+| Layout handles declared by modules     | New layout page handle                                                                                                   | MINOR |
+|                                        | New container/block added to handle                                                                                      | MINOR |
+|                                        | Removed layout handle                                                                                                    | MAJOR |
+| Static view files                      | JS/CSS/LESS file removed/moved                                                                                           | MAJOR |
+
+### Potential number of affected customisations
+
+In order to identify the volume of the impact the usages of the changed code should be detected in the available code/extensions/customisations.
+
+The analysis result should indicate the proportion of reviewed modules to the modules referencing (using) the changed code.
+
+### TCO Analysis result example
+
+The analysis result should indicate all the changes that have TCO impact (or simply state that there is "No TCO impact").
+
+For example, if the non-api "SomeClass::someMethod" was changed and the new required argument was added to the CLI command, the analysis result may look like:
+
+```text
+Changes have TCO impact.
+Change to SomeClass::someMethod is PATCH level but impacts 5 out of 10 reviewed custom modules.
+Change to CLI command is MAJOR level, it impacts 2 out of 10 reviewed custom modules.
+```
```

Comment:
Need to define guidelines for impact assessments: simply how to find those 5 of 10 and identify is it a high impact or low. Each extension has its own value for merchants with different sizes, while 4 of 10 may be ok with the changes they may be unacceptable for one certain XXL merchant.

Reply:
TCO results should contain all the information available to the person writing the result. Provided sample results are just examples.